agents/plugins/mk_mysql

```diff
@@ -27,11 +27,21 @@ function do_query() {
 
   echo "<<<mysql_slave>>>"
   echo "$INSTANCE_HEADER"
+  # do not fail but output nothing if we do not have a slave or do not have the required permission
   mysql --defaults-extra-file="$MK_CONFDIR"/mysql.cfg ${1:+--socket="$1"} -s \
-     -e "show slave status\G"
-
+     -e "show slave status\G" || :
 }
 
+# if mysqladmin is not retrievable via PATH try to receive mysql home from mysqld process
```

Comment:
Your proposed fix would be a major security issue. It would allow a regular user to just run a process by that name, and then have the agent execute something from the same directory as root.
I would be prepared to add some well-known folders as a fallback, but dynamically executing "anything" is not an option.

Reply:
That would be great if you are able to add some well-known folders as fallback. So the plugin is able to execute the mysql binaries used within that plugin.

Reply:
I'm fine with enhanceing the PATH environment variable with well known locations where mysqld will be installed or a option to add a directory to the PATH variable via WATO/Agent Bakery to the plugins configuration.

Reply:
Agent bakery itself does not support that very well currently. Can you instead go for option one and hard code some well known fall back folders into this PR?